.github/workflows/cicd.yml

```
@@ -13,73 +13,163 @@ concurrency:
+      needs.changes.outputs.should_skip != 'true' && (
+        needs.changes.outputs.any-code == 'true' ||
+        needs.changes.outputs.ci-config == 'true' ||
+        github.event_name != 'pull_request'
+      )
     secrets: inherit
     with:
-      should_skip: ${{ needs.pre-job.outputs.should_skip }}
+      should_skip: ${{ needs.changes.outputs.should_skip }}
+      # Force run all checks when CI config changes or on non-PR events (push to main, merge queue)
+      # This ensures workflow changes are fully validated
+      force_run: ${{ needs.changes.outputs.ci-config == 'true' || github.event_name != 'pull_request' }}
```

actionlint (1.7.9)

input "force_run" is typed as string by reusable workflow "./.github/workflows/ci-quality-gates.yml". bool value cannot be assigned

(expression)